<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>275</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>400</width>
    <height>275</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>400</width>
    <height>275</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QTextEdit" name="ReceiveText">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>400</width>
      <height>210</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="sizeIncrement">
     <size>
      <width>0</width>
      <height>0</height>
     </size>
    </property>
    <property name="autoFillBackground">
     <bool>true</bool>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLineEdit" name="SendText">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>210</y>
      <width>320</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="maxLength">
     <number>512</number>
    </property>
    <property name="placeholderText">
     <string>Message...</string>
    </property>
   </widget>
   <widget class="QPushButton" name="SendButton">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>210</y>
      <width>80</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Send</string>
    </property>
   </widget>
   <widget class="QWidget" name="ConnectSettingsWidget" native="true">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>400</width>
      <height>230</height>
     </rect>
    </property>
    <widget class="QLineEdit" name="PrimePText">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>30</y>
       <width>113</width>
       <height>25</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="PrimePLabel">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>30</y>
       <width>111</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Prime Number P: </string>
     </property>
    </widget>
    <widget class="QPushButton" name="PrimePRand">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>30</y>
       <width>80</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Random</string>
     </property>
    </widget>
    <widget class="QPushButton" name="PrimeQRand">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>70</y>
       <width>80</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Random</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="PrimeQText">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>70</y>
       <width>113</width>
       <height>25</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="PrimeQLabel">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>70</y>
       <width>111</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Prime Number Q: </string>
     </property>
    </widget>
    <widget class="QPushButton" name="EncKeyRand">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>110</y>
       <width>80</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Random</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="EncKeyText">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>110</y>
       <width>113</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>65537</string>
     </property>
    </widget>
    <widget class="QLabel" name="EncKeyLabel">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>110</y>
       <width>111</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Encryption Key: </string>
     </property>
    </widget>
    <widget class="QLabel" name="PeerIPLabel">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>190</y>
       <width>121</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Peer's IP Address: </string>
     </property>
    </widget>
    <widget class="QLineEdit" name="PeerIPText">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>190</y>
       <width>113</width>
       <height>25</height>
      </rect>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="maxLength">
      <number>15</number>
     </property>
    </widget>
    <widget class="QPushButton" name="ConnectButton">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>190</y>
       <width>80</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Connect</string>
     </property>
    </widget>
    <widget class="QLabel" name="DescLabel1">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>10</y>
       <width>191</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>RSA Encryption Values To Use...</string>
     </property>
    </widget>
    <widget class="QLabel" name="DescLabel2">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>170</y>
       <width>191</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>The IPv4 Address Of The Peer...</string>
     </property>
    </widget>
    <widget class="Line" name="ConnectLine">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>145</y>
       <width>400</width>
       <height>20</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="LoadMyKeysWidget" native="true">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>400</width>
      <height>230</height>
     </rect>
    </property>
    <widget class="QGroupBox" name="groupBox">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>381</width>
       <height>201</height>
      </rect>
     </property>
     <property name="title">
      <string>Set RSA Values From Files</string>
     </property>
     <widget class="QLineEdit" name="PasswordLine">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>90</y>
        <width>113</width>
        <height>23</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="OpenPublicButton">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>30</y>
        <width>80</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Open File</string>
      </property>
     </widget>
     <widget class="QLabel" name="PasswordLabel">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>90</y>
        <width>131</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Private Key Password:</string>
      </property>
     </widget>
     <widget class="QLabel" name="PrivateKeyLabel">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>120</y>
        <width>71</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Private Key: </string>
      </property>
     </widget>
     <widget class="QLabel" name="PublicKeyLabel">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>71</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Public Key: </string>
      </property>
     </widget>
     <widget class="QPushButton" name="OpenPrivateButton">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>120</y>
        <width>80</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Open File</string>
      </property>
     </widget>
     <widget class="QPushButton" name="OKButton">
      <property name="geometry">
       <rect>
        <x>290</x>
        <y>170</y>
        <width>81</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>OK</string>
      </property>
     </widget>
     <widget class="QLabel" name="PublicKeyLocLabel">
      <property name="geometry">
       <rect>
        <x>180</x>
        <y>30</y>
        <width>191</width>
        <height>50</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLabel" name="PrivateKeyLocLabel">
      <property name="geometry">
       <rect>
        <x>180</x>
        <y>120</y>
        <width>191</width>
        <height>50</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
     </widget>
     <zorder>PasswordLine</zorder>
     <zorder>PasswordLabel</zorder>
     <zorder>PrivateKeyLabel</zorder>
     <zorder>PublicKeyLabel</zorder>
     <zorder>OpenPrivateButton</zorder>
     <zorder>OKButton</zorder>
     <zorder>PublicKeyLocLabel</zorder>
     <zorder>PrivateKeyLocLabel</zorder>
     <zorder>OpenPublicButton</zorder>
    </widget>
   </widget>
   <widget class="QWidget" name="OptionsWidget" native="true">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>400</width>
      <height>230</height>
     </rect>
    </property>
    <widget class="QCheckBox" name="SavePublicCB">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>161</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Save Peer's Public Key</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="SendPublicCB">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>151</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Send Public Key</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>381</width>
       <height>151</height>
      </rect>
     </property>
     <property name="title">
      <string>Create Key Files</string>
     </property>
     <widget class="QLineEdit" name="MyPrivateLocLine">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>110</y>
        <width>101</width>
        <height>25</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>6</x>
        <y>30</y>
        <width>111</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Public Key Name</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>6</x>
        <y>80</y>
        <width>111</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Private Password</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>6</x>
        <y>110</y>
        <width>111</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Private Key Name</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="MyPublicLocLine">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>30</y>
        <width>101</width>
        <height>25</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="MyPrivatePassLine">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>80</y>
        <width>101</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QPushButton" name="CreateKeysButton">
      <property name="geometry">
       <rect>
        <x>260</x>
        <y>110</y>
        <width>80</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Create</string>
      </property>
     </widget>
    </widget>
    <widget class="QLineEdit" name="PeerPublicLocLine">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>10</y>
       <width>101</width>
       <height>25</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>40</y>
       <width>81</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Port Number</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="PortLine">
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>40</y>
       <width>101</width>
       <height>25</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>10</y>
       <width>81</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Public Name</string>
     </property>
    </widget>
   </widget>
   <widget class="QLabel" name="StatusLabel">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>234</y>
      <width>381</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Not Connected</string>
    </property>
   </widget>
   <zorder>ConnectSettingsWidget</zorder>
   <zorder>LoadMyKeysWidget</zorder>
   <zorder>OptionsWidget</zorder>
   <zorder>SendButton</zorder>
   <zorder>ReceiveText</zorder>
   <zorder>SendText</zorder>
   <zorder>StatusLabel</zorder>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>400</width>
     <height>20</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuCrypto_Chat">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionLoad_Keys"/>
    <addaction name="actionLoad_Peer_Public_Key"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionHelp"/>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menuAdvanced">
    <property name="title">
     <string>Advanced</string>
    </property>
    <addaction name="actionOptions"/>
   </widget>
   <widget class="QMenu" name="menuChat">
    <property name="title">
     <string>Chat</string>
    </property>
    <addaction name="actionConnect"/>
    <addaction name="actionDisconnect"/>
    <addaction name="separator"/>
    <addaction name="actionSend_File"/>
   </widget>
   <addaction name="menuCrypto_Chat"/>
   <addaction name="menuChat"/>
   <addaction name="menuAdvanced"/>
   <addaction name="menuHelp"/>
  </widget>
  <action name="actionLoad_Keys">
   <property name="text">
    <string>Load My Keys</string>
   </property>
  </action>
  <action name="actionLoad_Peer_Public_Key">
   <property name="text">
    <string>Load Peer Public Key</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionHelp">
   <property name="text">
    <string>Help</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionOptions">
   <property name="text">
    <string>Options</string>
   </property>
  </action>
  <action name="actionConnect">
   <property name="text">
    <string>Connect</string>
   </property>
  </action>
  <action name="actionSend_File">
   <property name="text">
    <string>Send File</string>
   </property>
  </action>
  <action name="actionDisconnect">
   <property name="text">
    <string>Disconnect</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
